feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/sendrequest/SendRequestFragment.java

```diff
@@ -94,72 +99,99 @@ public void onClick(View view) {
+                                            }
+                                        },null);
                             }
 
                         }
                     }, new VoidCallback() {
                         @Override
                         public void onCallback() {
-                            Toast toast = Toast.makeText(getActivity(), "User Not Exist, Invite your friend", Toast.LENGTH_LONG);
-                            toast.setGravity(Gravity.CENTER| Gravity.CENTER_HORIZONTAL, 0, 0);
-                            toast.show();
-                            usernameEditText.setText("");
+                            IndvalidUser();
                         }
                     });
+
                 }
             }
         });
         return root;
     }
 
+    public void AlreadyFollowedUser(){
+        Toast toast = Toast.makeText(getActivity(), "You have already followed this user", Toast.LENGTH_LONG);
+        toast.setGravity(Gravity.CENTER | Gravity.CENTER_HORIZONTAL, 0, 0);
+        toast.show();
+        usernameEditText.setText("");
+    }
 
-    /**
-     * Since Login/Register Activity haven't complete, so I use <filename> that I create manually. (which is Xiaole)
-     * Once sharedPreference create in the Login/Register Activity, I need to get the variable <file name>
-     * from mainActivity and pass the file name in SendRequestFragment --> TODO
-     *
-     * Login/Register should pass the filename variable to the MainActivity(Home Page).
-     */
+    public void AlreadySentRequest(){
+        Toast toast = Toast.makeText(getActivity(), "You have already sent request!", Toast.LENGTH_LONG);
+        toast.setGravity(Gravity.CENTER | Gravity.CENTER_HORIZONTAL, 0, 0);
+        toast.show();
+        usernameEditText.setText("");
+    }
 
+    public void ValidUser(){
+        Toast toast = Toast.makeText(getActivity(), "Valid User", Toast.LENGTH_LONG);
+        toast.setGravity(Gravity.CENTER | Gravity.CENTER_HORIZONTAL, 0, 0);
+        toast.show();
+    }
 
-//    private void storeRecipientUsername(String friendName){
-//        //TODO: pass filename variable from MainActivity
-//        SharedPreferences mySharedPreferences = this.getActivity().getSharedPreferences("Xiaole", Context.MODE_PRIVATE);
-//        SharedPreferences.Editor myEditor = mySharedPreferences.edit();
-//        myEditor.putString("recipientUsername", friendName);
-//        myEditor.apply();
-//    }
-//
-//    private String getRecipientUsername(){
-//        SharedPreferences mySharedPrefrences = this.getActivity().getSharedPreferences("Xiaole", Context.MODE_PRIVATE);
-//        //String RecipientUsername = mySharedPrefrences.getString("recipientUsername", this.getActivity().);
-//        String RecipientUsername = "Xiaole2";
-//        return RecipientUsername;
-//    }
+    public void IndvalidUser(){
+        Toast toast = Toast.makeText(getActivity(), "User Not Exist, Invite your friend", Toast.LENGTH_LONG);
```

Comment:
"This user _does_ not exist, invite your friends"

Reply:
fixed